process/block/preprocess/transactions.go

```diff
@@ -712,6 +718,20 @@ func (txs *transactions) computeOrderedTxs(
 	return orderedTxs, orderedTxHashes, nil
 }
 
+func getAlreadySortedTxs(provider SortedTransactionsProvider) ([]*transaction.Transaction, [][]byte) {
+	txs, hashes := provider.GetTransactions(process.MaxItemsInBlock, process.NumTxPerSenderBatchForFillingMiniblock)
+
+	txsAsStruct := make([]*transaction.Transaction, 0)
+	for _, tx := range txs {
```

Comment:
I do not like reconstruction these into new structures. You can do it only when needed, at processing.

Reply:
There still is some code that accepts the array of transactions as array of structures. I tried to touch as little code as possible (since the pull request is already large). Therefore, you will still see the function `castToTxsAsInterface`, but that is encapsulated in the adapter. Is that ok?

Reply:
Refactored, is it ok?